haptools/index.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+import logging
+from pathlib import Path
+from haptools.data.haplotypes import Haplotypes
+from pysam import tabix_index
+from haptools import data
+import tempfile
+
+"""
+
+    Used as a helper method for index_haps.
+    If an output path is not provided, the resulting file will have the same name as the input with .gz and .tbi appended to the appropriate files.
+
+    Parameters
+    ----------
+    
+    path : Path
+        The path to the haplotypes in a .hap file
+    suffix : str
+        The location to which to write output.  If an output location is not specified, the output will write to the same location as the input file.
+
+"""
+
+
+def append_suffix(
+    path: Path,
+    suffix: str,
+):
+    return path.with_suffix(path.suffix + suffix)
+
+
+"""
```

Comment:
yeah, this docstring one should also go underneath the function definition for `index_haps()`